### PR TITLE
[CALCITE-5759] 'SEARCH(1, Sarg[IS NOT NULL])' should be simplified to 'TRUE'

### PR DESCRIPTION
**Pull request by herunkang2018**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rex/RexProgramTest.java [hunk 1]
```diff
@@ -1768,6 +1768,68 @@ private void checkExponentialCnf(int n) {
     checkSimplify(expr, simplified);
   }
 
+  /** Unit test for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5759">[CALCITE-5759]
+   * SEARCH operator with special sarg is not fully simplified</a>. */
+  @Test void testSimplifySearchWithSpecialSargIsNotNull() {
+    RexNode intExpression =
```
Comment (libenchao) [inline core/src/test/java/org/apache/calcite/rex/RexProgramTest.java:1775]: How about name it `intLiteral`?

**Review by libenchao — COMMENTED** [review]
@herunkang2018 Thanks for your contribution, the changes looks great generally. I only have 1 nit comment.

Besides, for the commit message / Jira title, we can describe it more specifically, such as `SEARCH(1, Sarg[IS NOT NULL]) should be simplified to TRUE`, what do you think?

**Review by NobiGo — CHANGES_REQUESTED** [review]
LGTM. Only a small piece of advice. Need to add code annotation. Like others
`   // "x < x" simplifies to "false"`

File: core/src/test/java/org/apache/calcite/rex/RexProgramTest.java (hunk 1 above)
Comment (herunkang2018) [reply core/src/test/java/org/apache/calcite/rex/RexProgramTest.java:1775]: I will refine it.

**Comment by herunkang2018** [comment]
@libenchao @NobiGo Thanks a lot for reviewing.
I have changed the commit message / Jira title to `SEARCH(1, Sarg[IS NOT NULL]) should be simplified to TRUE` to make it more clear.

**Pull request closed without merging**